### ISSUE TITLE
Failed to load configuration with key workflow_dir in porla.toml file

### ISSUE THREAD

**Issue by paxter** [porla/porla#194]
If `workflow_dir` key is defined in config file, porla fails to start. After removing the key, everything is starting fine.

```
$ ./porla --config-file test.toml
[2023-06-16 17:33:18.577540] [0x0000ffff8c129008] [warning] No secret key set. Porla will generate one
[2023-06-16 17:33:18.577674] [0x0000ffff8c129008] [warning] Use './porla key:generate' to generate a secret key
[2023-06-16 17:33:18.631823] [0x0000ffff8c129008] [fatal]   Failed to load configuration
```

Used static binary from there: https://github.com/porla/porla/actions/runs/5242062776

**Comment by paxter** [comment]
Furthermore key `listen_interfaces` in porla.toml is ignored or not working properly. porla doesn't use set port.

```
listen_interfaces = [
  ["end0", 12345]
]
```

```
# netstat -tulpn | grep LISTEN         
tcp        0      0 127.0.0.1:6881          0.0.0.0:*               LISTEN      60657/porla         
tcp        0      0 192...:6881      0.0.0.0:*               LISTEN      60657/porla         
tcp        0      0 127.0.0.1:1337          0.0.0.0:*               LISTEN      60657/porla         
tcp6       0      0 ::1:6881                :::*                    LISTEN      60657/porla         
tcp6       0      0 fe80:::::fe9:6881 :::*                    LISTEN      60657/porla         
```

**Comment by vktr** [comment]
> If `workflow_dir` key is defined in config file, porla fails to start. After removing the key, everything is starting fine.
> 
> ```
> $ ./porla --config-file test.toml
> [2023-06-16 17:33:18.577540] [0x0000ffff8c129008] [warning] No secret key set. Porla will generate one
> [2023-06-16 17:33:18.577674] [0x0000ffff8c129008] [warning] Use './porla key:generate' to generate a secret key
> [2023-06-16 17:33:18.631823] [0x0000ffff8c129008] [fatal]   Failed to load configuration
> ```
> 
> Used static binary from there: https://github.com/porla/porla/actions/runs/5242062776

I can reproduce this only if I specify a workflow directory that does not exist. It shouldn't make Porla crash, however.

> Furthermore key `listen_interfaces` in porla.toml is ignored or not working properly. porla doesn't use set port.
> 
> ```
> listen_interfaces = [
>   ["end0", 12345]
> ]
> ```
> 
> ```
> # netstat -tulpn | grep LISTEN         
> tcp        0      0 127.0.0.1:6881          0.0.0.0:*               LISTEN      60657/porla         
> tcp        0      0 192...:6881      0.0.0.0:*               LISTEN      60657/porla         
> tcp        0      0 127.0.0.1:1337          0.0.0.0:*               LISTEN      60657/porla         
> tcp6       0      0 ::1:6881                :::*                    LISTEN      60657/porla         
> tcp6       0      0 fe80:::::fe9:6881 :::*                    LISTEN      60657/porla         
> ```

This, I cannot reproduce. Here is my `test.toml` file,

```toml
listen_interfaces = [
  ["en0", 19191]
]
```

And if I run `./porla --config-file test.toml` I can see that it listens on 19191 when running netstat.

**Comment by paxter** [comment]
Thanks for fast responding. You are right. The set folder does not exist. My fault.

Obviously `listen_interfaces` only works properly if no `session_settings` key is declared.

**Working**

```
listen_interfaces = [
  ["end0", 58741]
]

#[session_settings]
#base = "default" # (valid: "default", "high_performance" or "min_memory_usage")
```

```
$ ./porla --config-file test.toml --log-level=trace
[...]
[2023-06-16 18:30:38.745819] [0x0000ffff8ae42008] [trace]   Session alert: successfully listening on [TCP] 192.168.1.191:58741
[2023-06-16 18:30:38.745876] [0x0000ffff8ae42008] [trace]   Session alert: successfully listening on [uTP] 192.168.1.191:58741
[2023-06-16 18:30:38.745993] [0x0000ffff8ae42008] [trace]   Session alert: successfully listening on [TCP] [fe80::e65f:1ff:fe95:aad%end0]:58741
[2023-06-16 18:30:38.746048] [0x0000ffff8ae42008] [trace]   Session alert: successfully listening on [uTP] [fe80::e65f:1ff:fe95:aad%end0]:58741
[...]
```

**Failing**

```
listen_interfaces = [
  ["end0", 58741]
]

[session_settings]
base = "default" # (valid: "default", "high_performance" or "min_memory_usage")
```

```
$ ./porla --config-file test.toml --log-level=trace
[...]
[2023-06-16 18:32:24.145196] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [TCP] 127.0.0.1:6881
[2023-06-16 18:32:24.145252] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [uTP] 127.0.0.1:6881
[2023-06-16 18:32:24.145278] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [TCP] 192.168.1.191:6881
[2023-06-16 18:32:24.145301] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [uTP] 192.168.1.191:6881
[2023-06-16 18:32:24.145329] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [TCP] [::1]:6881
[2023-06-16 18:32:24.145354] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [uTP] [::1]:6881
[2023-06-16 18:32:24.145495] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [TCP] [fe80::e65f:1ff:fe95:aad%end0]:6881
[2023-06-16 18:32:24.145552] [0x0000ffffb3123008] [trace]   Session alert: successfully listening on [uTP] [fe80::e65f:1ff:fe95:aad%end0]:6881
[...]
```

**Comment by vktr** [comment]
The `workflow_dir` issue was fixed in #195.

Ah, yeah, I see now why the listen interfaces don't work. We apply the session settings base _after_ we set the listen interfaces. It should probably the other way around. I'll push a fix for that as well.

**Comment by paxter** [comment]
As usual. Thanks a lot. :)

**Comment by vktr** [comment]
Of course 😃 It will be fixed in #196.
